libsolidity/ast/AST.h

```diff
@@ -316,20 +316,24 @@ class ImplementationOptional
 class ContractDefinition: public Declaration, public Documented
 {
 public:
+	enum class ContractKind { Interface, Contract, Library };
+
 	ContractDefinition(
 		SourceLocation const& _location,
 		ASTPointer<ASTString> const& _name,
 		ASTPointer<ASTString> const& _documentation,
 		std::vector<ASTPointer<InheritanceSpecifier>> const& _baseContracts,
 		std::vector<ASTPointer<ASTNode>> const& _subNodes,
-		bool _isLibrary
+		ContractKind _contractKind = ContractKind::Contract
 	):
 		Declaration(_location, _name),
 		Documented(_documentation),
 		m_baseContracts(_baseContracts),
 		m_subNodes(_subNodes),
-		m_isLibrary(_isLibrary)
-	{}
+		m_contractKind(_contractKind)
+	{
+		m_isLibrary = (m_contractKind == ContractKind::Library);
```

Comment:
I'd prefer to do it, but it seems like quite an extensive change, which would hinder the review of this feature. I can prepare another PR changing that.